lib/api/src/backend/sys/entities/function/mod.rs

```diff
@@ -543,7 +543,19 @@ macro_rules! impl_host_function {
                 // See: https://github.com/wasmerio/wasmer/pull/5700
                 match result {
                     Ok(Ok(result)) => return result.into_c_struct(&mut store),
-                    Ok(Err(trap)) => raise_user_trap(trap),
+                    Ok(Err(trap)) => {
+                        // TODO: is it safe to do all this on the WASM stack?
```

Comment:
@syrusakbary please review this, I don't think it's safe to do it this way according to your comment above.

Reply:
@syrusakbary ping on this